### ISSUE TITLE
Unsuitable error message and status code when try to install plugin which doesn't respect minimum requirements.

### ISSUE THREAD

**Issue by miya0001** [wp-cli/extension-command#50]
The latest version of Akismet requires WordPres 4.0 or later.
Then we try to install Akismet to WordPress 3.7, error message will be unsuitable and status code will be `0`.

We need to improve output and it should return `1`.

```
$ wp plugin uninstall akismet
Uninstalled and deleted 'akismet' plugin.
Success: Uninstalled 1 of 1 plugins.
$ wp plugin install akismet --activate
Installing Akismet (4.0)
Downloading install package from https://downloads.wordpress.org/plugin/akismet.4.0.zip...
Using cached file '/home/ubuntu/.wp-cli/cache/plugin/akismet-4.0.zip'...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'akismet'...
<!doctype html>
<html>
<head>
<meta charset="UTF-8">
<style>
* {
	text-align: center;
	margin: 0;
	padding: 0;
	font-family: "Lucida Grande",Verdana,Arial,"Bitstream Vera Sans",sans-serif;
}
p {
	margin-top: 1em;
	font-size: 18px;
}
</style>
<body>
<p>&lt;strong&gt;Akismet 4.0 requires WordPress 4.0 or higher.&lt;/strong&gt; Please &lt;a href=&quot;https://codex.wordpress.org/Upgrading_WordPress&quot;&gt;upgrade WordPress&lt;/a&gt; to a current version, or &lt;a href=&quot;https://wordpress.org/extend/plugins/akismet/download/&quot;&gt;downgrade to version 2.4 of the Akismet plugin&lt;/a&gt;.</p>
</body>
</html>
$ echo $?
0
$ wp plugin list
+--------------------+----------+--------+---------+
| name               | status   | update | version |
+--------------------+----------+--------+---------+
| akismet            | inactive | none   | 4.0     |
| hello              | inactive | none   | 1.6     |
| wordpress-importer | inactive | none   | 0.6.3   |
| no-mail            | must-use | none   |         |
+--------------------+----------+--------+---------+
```

**Comment by miya0001** [comment]
Oh, do we need plugin which requires `WordPress 10.0` or later in the plugin repository? 😈

**Comment by miya0001** [comment]
Following tests didn't work as expected...
Maybe, minimum requires will be from plugins API.

```
  Scenario: Install plugin which doesn't support the WordPress version
    Given a WP install
    And I run `wp scaffold plugin very-future-plugin`
    And a wp-content/plugins/very-future-plugin/readme.txt file:
      """
      === Example Plugin ===
      Contributors: matt, ryan, andy
      Requires at least: 10.0
      Tested up to: 10.0
      Stable tag: 4.0
      License: GPLv2 or later
      """

    When I try `wp plugin activate very-future-plugin`
    Then the return code should be 1
```

**Comment by gitlost** [comment]
That's the Akismet `register_activation_hook()` printing that and exiting https://plugins.trac.wordpress.org/browser/akismet/trunk/class.akismet.php#L1113. So maybe we need some `register_shutdown_function()` mechanism to manage such cases better, changing the exit code or something.

**Comment by miya0001** [comment]
@gitlost 

Oh... true, I could install and activate wordpress-seo to WordPress 4.5.
(wordpress-seo requires WordPress 4.6)

```
$ wp core version
4.5
$ wp plugin install wordpress-seo --activate
Installing Yoast SEO (5.4.2)
Downloading install package from https://downloads.wordpress.org/plugin/wordpress-seo.5.4.2.zip...
Unpacking the package...
Installing the plugin...
Plugin installed successfully.
Activating 'wordpress-seo'...
Plugin 'wordpress-seo' activated.
Success: Installed 1 of 1 plugins.
$ wp plugin list
+---------------+----------+-----------+---------+
| name          | status   | update    | version |
+---------------+----------+-----------+---------+
| akismet       | inactive | available | 3.1.10  |
| hello         | inactive | none      | 1.6     |
| wordpress-seo | active   | none      | 5.4.2   |
| no-mail       | must-use | none      |         |
+---------------+----------+-----------+---------+
```

Also, I could Install and activate it from dashboard.
![](https://www.evernote.com/l/ABXr1D0kvzpCda_zRhcg_4J48U0sPNkt5xAB/image.png)

I think it should be handled on WordPress core, so I would close this issue.

Thanks 😊
